src/main/java/build/buildfarm/instance/shard/RedisShardSubscriber.java

```diff
@@ -232,13 +235,16 @@ void onWorkerChange(WorkerChange workerChange) {
   }
 
   void addWorker(WorkerChange workerChange) {
-    synchronized (workers) {
-      workers.put(
-          workerChange.getName(),
-          ShardWorker.newBuilder()
-              .setEndpoint(workerChange.getName())
-              .setFirstRegisteredAt(Timestamps.toMillis(workerChange.getAdd().getEffectiveAt()))
-              .build());
+    if ((workerChange.getAdd().getWorkerType() & workerType) > 0) {
```

Comment:
!= 0 here please - nothing saying that the mask has to be in non-negative range.

Reply:
Done